src/Form/index.tsx

```diff
@@ -959,29 +961,78 @@ function Form({
   // usePayments (Stripe)
   const [getCardElement, setCardElement] = usePayments();
 
-  async function handleStepSubmissionPayment(formattedFields: any) {
+  async function setupStepPaymentMethod(formattedFields: any) {
     for (let i = 0; i < activeStep.servar_fields.length; i++) {
       const servar = activeStep.servar_fields[i].servar;
       if (servar.type === 'payment_method') {
         const integrationData = integrations?.stripe;
         const actionData: ActionData = {
-          fieldVal: fieldValues[servar.key],
           servar,
           client,
           formattedFields,
           updateFieldValues,
-          step: activeStep,
           integrationData,
           targetElement: getCardElement(servar.key)
         };
-        const actionResult = await setupPaymentMethodAndPay(actionData);
+        const actionResult = await setupPaymentMethod(actionData);
         return actionResult ?? {};
       }
     }
-
     return {};
   }
 
+  async function collectPaymentAction(
+    triggerElement: any,
+    setLoader = () => {}
+  ) {
+    const errorCallback = getErrorCallback({
+      // could be container or button but model as button for the time being...
+      trigger: lookUpTrigger(activeStep, triggerElement.id, 'button')
+    });
+    // validate all step fields and buttons.  Must be valid before payment.
+    const { invalid, inlineErrors: newInlineErrors } = validateElements({
+      elements: [...activeStep.servar_fields, ...activeStep.buttons],
+      triggerErrors: true,
+      errorType: formSettings.errorType,
+      formRef,
+      errorCallback,
+      setInlineErrors
+    });
+    if (invalid) return false;
+
+    // payment/checkout
+    const pm = activeStep.servar_fields.find(
+      ({ servar: { type } }: any) => type === 'payment_method'
+    );
+    setLoader();
```

Comment:
loader logic doesn't need to exist here anymore. loaders are set if configured before all actions are run and cleared after they run